AI #1336: Refactor entity name references and usage

rileyjenk opened the pull request:
This PR attempts to consistently apply usage of the following entity identification terms:

* `service provider`
* `host provider`
* `data generator`
* `invoice issuer`

rileyjenk committed:
Realigning terminology to the new entity definitions.

rileyjenk committed:
updating csv examples to have new column

rileyjenk committed:
typo fix

rileyjenk committed:
Fixing column list

rileyjenk commented:
Ok a few call outs here:

- I have attempted to fix all the references where we have used provider(many) or publisher(almost none)
- We have brand new feature that I have...slightly...renamed (provider-calculated-split-cost-allocation is now service-provider-calculated-split-cost-allocation)
- I have updated the csv examples...which are rather sparse on the other columns.

rileyjenk committed:
extra space

rileyjenk commented on specification/datasets/cost_and_usage/columns/availabilityzone.md:
```diff
@@ -1,10 +1,10 @@
 The AvailabilityZone column adheres to the following requirements:
 
-* AvailabilityZone is RECOMMENDED to be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports deploying resources or services within an *availability zone*.
+* AvailabilityZone is RECOMMENDED to be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the data generator supports deploying resources or services within an *availability zone*.
```
Fix this it needs to be host-provider

rileyjenk replied:
```suggestion
* AvailabilityZone is RECOMMENDED to be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host-provider supports deploying resources or services within an *availability zone*.
```

shawnalpay replied:
```suggestion
* AvailabilityZone is RECOMMENDED to be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host provider supports deploying resources or services within an *availability zone*.
```

rileyjenk commented on specification/datasets/cost_and_usage/columns/regionid.md:
```diff
@@ -1,10 +1,10 @@
 # Region ID
 
-A Region ID is a provider-assigned identifier for an isolated geographic area where a [*resource*](#glossary:resource) is provisioned or a [*service*](#glossary:service) is provided. The region is commonly used for scenarios like analyzing cost and unit prices based on where *resources* are deployed.
+A Region ID is a service-provider-assigned identifier for an isolated geographic area where a [*resource*](#glossary:resource) is provisioned or a [*service*](#glossary:service) is provided. The region is commonly used for scenarios like analyzing cost and unit prices based on where *resources* are deployed.
```
```suggestion
A Region ID is a host-provider-assigned identifier for an isolated geographic area where a [*resource*](#glossary:resource) is provisioned or a [*service*](#glossary:service) is provided. The region is commonly used for scenarios like analyzing cost and unit prices based on where *resources* are deployed.
```

shawnalpay replied:
Looks like this was applied.  Resolving.

rileyjenk commented on specification/datasets/cost_and_usage/columns/regionid.md:
```diff
@@ -1,10 +1,10 @@
 The RegionId column adheres to the following requirements:
 
-* RegionId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports deploying resources or services within a region.
+* RegionId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the service provider supports deploying resources or services within a region.
```
```suggestion
* RegionId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host provider supports deploying resources or services within a region.
```

shawnalpay replied:
Looks like this was applied.  Resolving.

rileyjenk commented on specification/datasets/cost_and_usage/columns/regionid.md:
```diff
@@ -21,7 +21,7 @@ Region ID
 ## Description
 
-Provider-assigned identifier for an isolated geographic area where a *resource* is provisioned or a *service* is provided.
+Service-provider-assigned identifier for an isolated geographic area where a *resource* is provisioned or a *service* is provided.
```
```suggestion
Host-provider-assigned identifier for an isolated geographic area where a *resource* is provisioned or a *service* is provided.
```

shawnalpay replied:
Looks like this was applied.  Resolving.

shawnalpay commented on guidelines/spec-design-guidelines.md:
```diff
@@ -7,8 +7,8 @@ Specification Content
-* must not specify that a specific technology, software package/library, or cloud provider must be (or is recommended to be) used
-* can use examples that show a specific technology, software package/library, or cloud provider to demonstrate a concept
+* must not specify that a specific technology, software package/library, or cloud service provider must be (or is recommended to be) used
+* can use examples that show a specific technology, software package/library, or cloud servie provider to demonstrate a concept
```
```suggestion
* must not specify that a specific technology, software package/library, or service provider must be (or is recommended to be) used
* can use examples that show a specific technology, software package/library, or service provider to demonstrate a concept
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/billingaccountid.md:
```diff
@@ -1,14 +1,14 @@
 * BillingAccountId MUST conform to [StringHandling](#stringhandling) requirements.
 * BillingAccountId MUST NOT be null.
-* BillingAccountId MUST be a unique identifier within a provider.
+* BillingAccountId MUST be a unique identifier within a invoice issuer.
```
```suggestion
* BillingAccountId MUST be a unique identifier within an invoice issuer.
```

rileyjenk commented:
Resource...need to look at that.

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/billingaccountid.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -10,21 +10,21 @@ The FOCUS-defined properties are:
 The AllocatedMethodDetails column adheres to the following requirements:
 
-* AllocatedMethodDetails SHOULD be present in a cost and usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports [Provider-Calculated Split Cost Allocation](#ProviderCalculatedSplitCostAllocation).
+* AllocatedMethodDetails SHOULD be present in a cost and usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the service provider supports [Provider-Calculated Split Cost Allocation](#ProviderCalculatedSplitCostAllocation).
```
```suggestion
* AllocatedMethodDetails SHOULD be present in a cost and usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the service provider supports [Service-Provider-Calculated Split Cost Allocation](#ServiceProviderCalculatedSplitCostAllocation).
```

rileyjenk replied:
Data Generator is the vote

shawnalpay replied:
Addressed via separate commit.  @rileyjenk please review and confirm it looks good.

shawnalpay replied:
Just to confirm: this ask is still valid after the merge.  You should look through and confirm I did this correctly across all files.

rileyjenk replied:
LGTM

shawnalpay replied:
Thanks!  Resolving.

shawnalpay commented on specification/appendix/saas_examples/saas_examples.mdpp:
```diff
@@ -1,6 +1,6 @@
 # Examples: SaaS
 
-The following section contains examples with how SaaS providers may implement the FOCUS specification. SaaS provider implementations will vary on the level of the detail available in their data, contract terms, purchasing options, and other factors.
+The following section contains examples with how SaaS Data Generators may implement the FOCUS specification. SaaS data generators implementations will vary on the level of the detail available in their data, contract terms, purchasing options, and other factors.
```
```suggestion
The following section contains examples with how SaaS data generators may implement the FOCUS specification. SaaS data generator implementations will vary on the level of the detail available in their data, contract terms, purchasing options, and other factors.
```

shawnalpay committed:
changing all references of split cost allocation to use "data generator"

shawnalpay committed:
revise anchor link reference to datagenerator-calculatedsplitcostallocation

shawnalpay commented on guidelines/data_model_design_guidelines.md:
```diff
@@ -108,4 +108,4 @@ The FOCUS specification aims “‘to create a standardized schema for cloud bil
 ### Example
 
-A new dataset for usage metadata should use vendor-neutral terminology and avoid provider-specific fields that could fragment the standard.
+A new dataset for usage metadata should use vendor-neutral terminology and avoid service-provider-specific fields that could fragment the standard.
```
```suggestion
A new dataset for usage metadata should use vendor-neutral terminology and avoid fields specific to a given service provider or data generator that could fragment the standard.
```

shawnalpay commented on guidelines/development-processes.md:
```diff
@@ -44,10 +44,10 @@ All suggestions and feature requests should be introduced to the FOCUS project v
 - **Start with a standard verb**. Choose one or use a verb similar to: Add, Clarify, Standardize, Enable, Define, Remove, Rename
 - **Use correct sentence structure**. Pick whichever reads best between: Verb + Target + Qualifier OR Verb + Qualifier + Target
-(e.g. “Add provider-calculated shared cost allocation” vs “Standardize tag export across clouds”)
+(e.g. “Add service-provider-calculated shared cost allocation” vs “Standardize tag export across clouds”)
```
```suggestion
(e.g. “Add data generator-calculated shared cost allocation” vs “Standardize tag export across clouds”)
```

shawnalpay commented on guidelines/normative-requirements-guidelines.md:
```diff
@@ -132,7 +132,7 @@
 #### Key-Value Pairs
 
-* **References to Key-Value Pairs depend on the context**: The terminology for key-value pairs varies depending on the column and context. For instance, when referring to key-value pairs, **tags**, **user-defined tags**, and **provider-defined tags** are used in **Tags**, whereas **SkuPriceDetails property** is used in **SkuPriceDetails**.
+* **References to Key-Value Pairs depend on the context**: The terminology for key-value pairs varies depending on the column and context. For instance, when referring to key-value pairs, **tags**, **user-defined tags**, and **data generator-defined tags** are used in **Tags**, whereas **SkuPriceDetails property** is used in **SkuPriceDetails**.
```
This section needs harmonization with rows 388 and 389 (see below).  Should it be data generator or service provider?  I vote for data generator.

shawnalpay commented on guidelines/normative-requirements-guidelines.md:
```diff
@@ -385,8 +385,8 @@ To ensure clarity and consistency across columns and requirements, it is importa
 * Examples:
-  * When the provider has only one user-defined tag scheme. (instead of: When the provider has only 1 user-defined tag scheme.)
-  * When the provider has more than one user-defined tag scheme. (instead of: When the provider has 2 or more user-defined tag schemes.)
+  * When the service provider has only one user-defined tag scheme. (instead of: When the service provider has only 1 user-defined tag scheme.)
+  * When the service provider has more than one user-defined tag scheme. (instead of: When the service provider has 2 or more user-defined tag schemes.)
```
This section needs harmonization with rows 135 (see above).  Should it be data generator or service provider?  I vote for data generator.

shawnalpay commented on guidelines/writing-requirements-model-guidelines.md:
```diff
@@ -43,14 +43,14 @@ The first stage of conversion of rules from the normative text to model rules is
 #### Flow Diagram
 
-The **Flow Diagram** in Stage 1 illustrates the end-to-end process used to extract Requirements Model from the FOCUS Specification. It begins by identifying the structural or conceptual entity to which each rule applies—such as Dataset, Row, Column, Attribute, or Metadata—and then guides users through a standardized sequence of steps to assign a unique identifier, classify the rule type, determin ...
+The **Flow Diagram** in Stage 1 illustrates the end-to-end process used to extract Requirements Model from the FOCUS Specification. It begins by identifying the structural or conceptual entity to which each rule applies—such as Dataset, Row, Column, Attribute, or Metadata—and then guides users through a standardized sequence of steps to assign a unique identifier, classify the rule type, determin ...
```
```suggestion
The **Flow Diagram** in Stage 1 illustrates the end-to-end process used to extract Requirements Model from the FOCUS Specification. It begins by identifying the structural or conceptual entity to which each rule applies—such as Dataset, Row, Column, Attribute, or Metadata—and then guides users through a standardized sequence of steps to assign a unique identifier, classify the rule type, determine conditional logic, and express validation criteria. This structured workflow ensures that all normative requirements from the specification are captured in a consistent, testable, and programmatically analyzable format. The diagram also reflects the relationships between FOCUS architectural components and highlights how entities like Service Provider Name influence rule applicability.
```

shawnalpay commented on guidelines/writing-requirements-model-guidelines.md:
```diff
@@ -43,14 +43,14 @@ The first stage of conversion of rules from the normative text to model rules is
 #### 1. Target Entity – Determine the entity
 
-Identify the target for the rule: **Dataset**, **Column**, **Attribute** property, **Provider**, etc. This sets the scope of the model requirement.
+Identify the target for the rule: **Dataset**, **Column**, **Attribute** property, **Service Provider**, etc. This sets the scope of the model requirement.
```
```suggestion
Identify the target for the rule: **Dataset**, **Column**, **Attribute** property, **Service Provider Name**, etc. This sets the scope of the model requirement.
```

shawnalpay commented on guidelines/writing-requirements-model-guidelines.md:
```diff
@@ -43,14 +43,14 @@ The first stage of conversion of rules from the normative text to model rules is
 
-The **Flow Diagram** in Stage 1 illustrates the end-to-end process used to extract Requirements Model from the FOCUS Specification. It begins by identifying the structural or conceptual entity to which each rule applies—such as Dataset, Row, Column, Attribute, or Metadata—and then guides users through a standardized sequence of steps to assign a unique identifier, classify the rule type, determin ...
+The **Flow Diagram** in Stage 1 illustrates the end-to-end process used to extract Requirements Model from the FOCUS Specification. It begins by identifying the structural or conceptual entity to which each rule applies—such as Dataset, Row, Column, Attribute, or Metadata—and then guides users through a standardized sequence of steps to assign a unique identifier, classify the rule type, determin ...
 <img width="812" height="1036" alt="Image" src="https://github.com/user-attachments/assets/040ba557-cc42-402b-b52a-742b88e40d54" />
```
@jpradocueva This diagram needs updating to reference Service Provider, not just Provider.

jpradocueva replied:
The diagram has been updated to reflect "Service Provider"

shawnalpay commented on guidelines/writing-requirements-model-guidelines.md:
```diff
@@ -94,20 +94,20 @@ Provider -.->|Used in rule conditions| Column
+| `Column`           | Named field across rows                             | Data type, format, constraints            | Column `BillingPeriodStart` MUST be of type `DateTime`                                                          |
+| `Attribute`        | Shared formatting/logic constraint                  | Formatting consistency across columns     | All `String` columns MUST conform to `StringHandling` requirements                                              |
+| `Metadata`         | Schema-level dataset descriptors                    | Schema versioning, declaration            | Metadata MUST declare `focus_version` as a valid semantic version string (e.g., "1.2.0")                        |
+| `Service Provider` | Entity that made the resource availabel for purchase | Conditional logic in requirements         | Column `CapacityReservationId` MUST be present when the service provider supports capacity reservation features |
```
```suggestion
| `Service Provider` | Entity that made the resource available for purchase | Conditional logic in requirements         | Column `CapacityReservationId` MUST be present when the service provider supports capacity reservation features |
```

shawnalpay commented on specification/appendix/saas_examples/simple_agreements.md:
```diff
@@ -32,7 +32,7 @@ Note the following details in the example dataset:
 * The single charge provided includes a ListUnitPrice for the licenses and a Pricing Quantity.
 
-## Scenario A3: Additional Purchase Records Provided in the SaaS Provider's FOCUS Data
+## Scenario A3: Additional Purchase Records Provided in the SaaS Service Provider's FOCUS Data
```
```suggestion
## Scenario A3: Additional Purchase Records Provided in the SaaS Data Generator's FOCUS Dataset
```

shawnalpay commented on specification/appendix/saas_examples/virtual_currency_pricing_model.md:
```diff
@@ -1,14 +1,14 @@
 The following baseline conditions apply to the scenarios described below:
 
-* AwesomeCorp has signed an agreement with SaaS provider Acme Co to use their services.
+* AwesomeCorp has signed an agreement with SaaS servie provider Acme Co to use their services.
```
```suggestion
* AwesomeCorp has signed an agreement with SaaS service provider Acme Co to use their services.
```

shawnalpay committed:
missed one split cost reference

shawnalpay commented on specification/attributes/invoice_handling.md:
```diff
@@ -19,14 +19,14 @@ Indicates how invoice-level *charges*, including those not directly tied to usag
 
 * All costs that appear on any invoice issued to a [*BillingAccountId*](#billingaccountid) MUST be included in the FOCUS Cost and Usage dataset.
-* If an invoice-level *charge* appears on a customer invoice but cannot be expressed using existing FOCUS columns, providers MUST include provider-defined columns (e.g., x_ChargeSubType) to capture the non-FOCUS-defined details needed to support invoice *charges* reconciliation using the FOCUS Cost and Usage dataset.
+* If an invoice-level *charge* appears on a customer invoice but cannot be expressed using existing FOCUS columns, service providers MUST include service-provider-defined columns (e.g., x_ChargeSubType) to capture the non-FOCUS-defined details needed to support invoice *charges* reconciliation using the FOCUS Cost and Usage dataset.
```
```suggestion
* If an invoice-level *charge* appears on a customer invoice but cannot be expressed using existing FOCUS columns, data generators MUST include data generator-defined columns (e.g., x_ChargeSubType) to capture the non-FOCUS-defined details needed to support invoice *charges* reconciliation using the FOCUS Cost and Usage dataset.
```

shawnalpay commented on specification/attributes/invoice_handling.md:
```diff
@@ -19,14 +19,14 @@ Indicates how invoice-level *charges*, including those not directly tied to usag
   * Tax exemption notifications
   * SLA credit details when the credit is already applied to the charged amount
-* If such informational items are excluded, providers MUST document this in their FOCUS implementation guide and ensure the sum of included charges still equals the invoice total.
+* If such informational items are excluded, service providers MUST document this in their FOCUS implementation guide and ensure the sum of included charges still equals the invoice total.
```
```suggestion
* If such informational items are excluded, data generators MUST document this in their FOCUS implementation guide and ensure the sum of included charges still equals the invoice total.
```

shawnalpay commented on specification/attributes/string_handling.md:
```diff
@@ -18,7 +18,7 @@ Requirements for string-capturing columns appearing in a [*FOCUS dataset*](#glos
 ## Requirements
 
-* String values MUST maintain the original casing, spacing, and other relevant consistency factors as specified by providers and end-users.
+* String values MUST maintain the original casing, spacing, and other relevant consistency factors as specified by service providers and end-users.
```
```suggestion
* String values MUST maintain the original casing, spacing, and other relevant consistency factors as specified by data generators and end-users.
```

shawnalpay committed:
a bunch more split cost allocation references fixed to leverage data generator

shawnalpay committed:
and one more reference

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -38,13 +38,13 @@ When AllocatedMethodDetails is not null, the JsonObjectFormat for AllocatedMetho
       * Allocation property value MUST be of the type specified for that property.
       * Allocation properties MUST adhere to additional normative requirements specific to that property.
-    * Provider-defined allocation properties MAY be included in "Elements".
+    * Service-provider-defined allocation properties MAY be included in "Elements".
```
```suggestion
    * Data generator-defined allocation properties MAY be included in "Elements".
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -38,13 +38,13 @@ When AllocatedMethodDetails is not null, the JsonObjectFormat for AllocatedMetho
+    * Service-provider-defined allocation properties MAY be included in "Elements".
       * Allocation property keys MUST begin with the string "x_" unless it is a FOCUS-defined allocation property.
-* AllocatedMethodDetails root object MAY contain additional provider-defined items, in addition to "Elements".
+* AllocatedMethodDetails root object MAY contain additional service-provider-defined items, in addition to "Elements".
```
```suggestion
* AllocatedMethodDetails root object MAY contain additional data generator-defined items, in addition to "Elements".
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -38,13 +38,13 @@ When AllocatedMethodDetails is not null, the JsonObjectFormat for AllocatedMetho
 ### Content Requirements
 
-The following keys are used for allocation properties to facilitate querying data across allocations and across providers. Focus-defined keys will appear in the list below and Provider-defined keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
+The following keys are used for allocation properties to facilitate querying data across allocations and across service providers. Focus-defined keys will appear in the list below and Service-provider-defined keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
```
```suggestion
The following keys are used for allocation properties to facilitate querying data across allocations and across data generators. Focus-defined keys will appear in the list below and data generator-defined keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -63,7 +63,7 @@ Usage Unit communicates the aspect of the documented Allocation Method Id being
 
 * "UsageUnit" MUST be included inside an "Elements" object if "UsageQuantity" allocation property is included in that "Elements" object, otherwise "UsageUnit" MAY be included in each "Elements" object.
-* Values for "UsageUnit" MUST capture the unit or component of provider's documented [AllocationMethod](#allocationmethodid) that was used to determine the "AllocatedRatio" value.
+* Values for "UsageUnit" MUST capture the unit or component of service provider's documented [AllocationMethod](#allocationmethodid) that was used to determine the "AllocatedRatio" value.
```
```suggestion
* Values for "UsageUnit" MUST capture the unit or component of data generator's documented [AllocationMethod](#allocationmethodid) that was used to determine the "AllocatedRatio" value.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -74,7 +74,7 @@ The "UsageQuantity" property adheres to the following requirements:
 * "UsageQuantity" MAY be included inside an "Elements" object when that "Elements" object contains a "UsageUnit" allocation property.
 * Values for "UsageQuantity" MUST be compatible with NumericFormat.
-* Values for "UsageQuantity" SHOULD capture the quantity or volume of the "UsageUnit" measured by the provider that was used to determine the "AllocatedRatio" value.
+* Values for "UsageQuantity" SHOULD capture the quantity or volume of the "UsageUnit" measured by the service provider that was used to determine the "AllocatedRatio" value.
```
```suggestion
* Values for "UsageQuantity" SHOULD capture the quantity or volume of the "UsageUnit" measured by the data generator that was used to determine the "AllocatedRatio" value.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
````diff
@@ -170,9 +170,9 @@ When multiple "UsageUnit" values are used to calculate the allaction, another ob
 }
 ```
-### Scenario 3: Provider omits keys that are not required
+### Scenario 3: Service provider omits keys that are not required
````
```suggestion
### Scenario 3: Data generator omits keys that are not required
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -170,9 +170,9 @@ When multiple "UsageUnit" values are used to calculate the allaction, another ob
+### Scenario 3: Service provider omits keys that are not required
 
-This provider does not wish to supply the "UsageUnit" or "UsageQuantity" keys but still provides cost allocation with some additional allocation method details. In this case, "UsageUnit" and "UsageQuantity" are omitted, and only the "AllocatedRatio" is supplied.
+This service provider does not wish to supply the "UsageUnit" or "UsageQuantity" keys but still provides cost allocation with some additional allocation method details. In this case, "UsageUnit" and "UsageQuantity" are omitted, and only the "AllocatedRatio" is supplied.
```
```suggestion
This data generator does not wish to supply the "UsageUnit" or "UsageQuantity" keys but still provides cost allocation with some additional allocation method details. In this case, "UsageUnit" and "UsageQuantity" are omitted, and only the "AllocatedRatio" is supplied.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -184,7 +184,7 @@ This provider does not wish to supply the "UsageUnit" or "UsageQuantity" keys bu
 ### Scenario 4: Additional non-FOCUS specified properties
 
-A provider can add additional properties if they feel more context is helpful or neccessary to the practitioner. In this scenario,the provider is supplying additional context that shows only 0.5 of a unit was used. However, since 1 unit was requested by the service this allocation represents, the allocation is being charged at 1 regardless.
+A service provider can add additional properties if they feel more context is helpful or neccessary to the practitioner. In this scenario,the service provider is supplying additional context that shows only 0.5 of a unit was used. However, since 1 unit was requested by the service this allocation represents, the allocation is being charged at 1 regardless.
```
```suggestion
A data generator can add additional properties if they feel more context is helpful or necessary to the practitioner. In this scenario, the data generator is supplying additional context that shows only 0.5 of a unit was used. However, since 1 unit was requested by the service this allocation represents, the allocation is being charged at 1 regardless.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethoddetails.md:
```diff
@@ -210,7 +210,7 @@ Allocated Method Details
 ## Description
 
-A set of properties describing how resources are allocated in provider-defined split cost allocation.
+A set of properties describing how resources are allocated in service-provider-defined split cost allocation.
```
```suggestion
A set of properties describing how resources are allocated in data generator-defined split cost allocation.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethodid.md:
```diff
@@ -1,16 +1,16 @@
 The AllocatedMethodId column adheres to the following requirements:
 
-* AllocatedMethodId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports provider-calculated split cost allocation.
+* AllocatedMethodId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the servie provider supports data generator-calculated split cost allocation.
```
```suggestion
* AllocatedMethodId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the data generator supports data generator-calculated split cost allocation.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedmethodid.md:
```diff
@@ -1,16 +1,16 @@
-* Provider documentation of a split cost allocation method MUST make reference to a single AllocatedMethodId value.
+  * AllocatedMethodId MUST be null when a [*charge*](#glossary:charge) is not related to a data generator-calculated split cost allocation.
+  * AllocatedMethodId MUST NOT be null when a *charge* is related to a data generator-calculated split cost allocation.
+* Service provider documentation of a split cost allocation method MUST make reference to a single AllocatedMethodId value.
```
```suggestion
* Data generator documentation of a split cost allocation method MUST make reference to a single AllocatedMethodId value.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedresourceid.md:
```diff
@@ -1,14 +1,14 @@
 # Allocated Resource ID
 
-An Allocated Resource ID is an identifier assigned by the provider which cost is being allocated to in a [Provider-Calculated Split Cost Allocation](#providercalculatedsplitcosthandling). The Allocated Resource ID is used to understand what the cost is being allocated to in [*charges*](#glossary:charge) where the provider is allocating costs to something other than the *charge's* [ResourceID](#Re ...
+An Allocated Resource ID is an identifier assigned by the service provider which cost is being allocated to in a [Data Generator-Calculated Split Cost Allocation](#serviceprovidercalculatedsplitcosthandling). The Allocated Resource ID is used to understand what the cost is being allocated to in [*charges*](#glossary:charge) where the service provider is allocating costs to something other than th ...
```
```suggestion
An Allocated Resource ID is an identifier assigned by the data generator which cost is being allocated to in a [Data Generator-Calculated Split Cost Allocation](#serviceprovidercalculatedsplitcosthandling). The Allocated Resource ID is used to understand what the cost is being allocated to in [*charges*](#glossary:charge) where the data generator is allocating costs to something other than the *charge's* [ResourceID](#ResourceId), as is the case for [allocated charges](#glossary:allocated-charge).
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
 * When AllocatedTags is not null, AllocatedTags adheres to the following additional requirements:
   * AllocatedTags MUST NOT include resource tags already present in [Tags](#tags).
-  * AllocatedTags MUST include all applicable user-defined and provider-defined tags for the [AllocatedResourceId](#AllocatedResourceId).
+  * AllocatedTags MUST include all applicable user-defined and service-provider-defined tags for the [AllocatedResourceId](#AllocatedResourceId).
```
```suggestion
  * AllocatedTags MUST include all applicable user-defined and data generator-defined tags for the [AllocatedResourceId](#AllocatedResourceId).
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
+  * Service provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
+* Service-provider-defined tags adhere to the following additional requirements:
+  * Service-provider-defined tag keys MUST be prefixed with a predetermined, service-provider-specified tag key prefix that is unique to each corresponding provider-specified tag scheme.
+  * Service provider SHOULD publish all service-provider-specified tag key prefixes within their respective documentation.
```
```suggestion
  * Data generator MUST NOT alter tag values unless applying true (boolean) to valueless tags.
* Data generator-defined tags adhere to the following additional requirements:
  * Data generator-defined tag keys MUST be prefixed with a predetermined, data generator-specified tag key prefix that is unique to each corresponding provider-specified tag scheme.
  * Data generator SHOULD publish all data generator-specified tag key prefixes within their respective documentation.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
+  * Service provider SHOULD publish all service-provider-specified tag key prefixes within their respective documentation.
 * User-defined tags adhere to the following additional requirements:
-  * Provider MUST prefix all user-defined tags scheme with a predetermined, provider-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the provider has more than one user-defined tag scheme.
+  * Service provider MUST prefix all user-defined tags scheme with a predetermined, service-provider-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the service-provider has more than one user-defined tag scheme.
```
```suggestion
  * Data generator MUST prefix all user-defined tags scheme with a predetermined, data generator-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the data generator has more than one user-defined tag scheme.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
+  * Service provider MUST prefix all user-defined tags scheme with a predetermined, service-provider-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the service-provider has more than one user-defined tag scheme.
 
-## Provider-Defined vs. User-Defined Tags
+## Service-Provider-Defined vs. User-Defined Tags
```
```suggestion
## Data Generator-Defined vs. User-Defined Tags
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
+## Service-Provider-Defined vs. User-Defined Tags
 
-This example illustrates various tags produced from multiple user-defined and provider-defined tag schemes. The first two tags illustrate examples from two different, user-defined tag schemes. The second tag is produced from a valueless, user-defined tag scheme, so the provider also applies `true` as its default value.
+This example illustrates various tags produced from multiple user-defined and service-provider-defined tag schemes. The first two tags illustrate examples from two different, user-defined tag schemes. The second tag is produced from a valueless, user-defined tag scheme, so the service provider also applies `true` as its default value.
```
```suggestion
This example illustrates various tags produced from multiple user-defined and data generator-defined tag schemes. The first two tags illustrate examples from two different, user-defined tag schemes. The second tag is produced from a valueless, user-defined tag scheme, so the data generator also applies `true` as its default value.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/allocatedtags.md:
```diff
@@ -1,30 +1,30 @@
+This example illustrates various tags produced from multiple user-defined and service-provider-defined tag schemes. The first two tags illustrate examples from two different, user-defined tag schemes. The second tag is produced from a valueless, user-defined tag scheme, so the service provider also applies `true` as its default value.
 
-The last two tags illustrate examples from two different, provider-defined tag schemes.
+The last two tags illustrate examples from two different, service-provider-defined tag schemes.
```
```suggestion
The last two tags illustrate examples from two different, data generator-defined tag schemes.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/regionname.md:
```diff
@@ -1,10 +1,10 @@
 The RegionName column adheres to the following requirements:
 
-* RegionName MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports deploying resources or services within a region.
+* RegionName MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host-provider supports deploying resources or services within a region.
```
```suggestion
* RegionName MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host provider supports deploying resources or services within a region.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/servicesubcategory.md:
```diff
@@ -85,7 +85,7 @@ Allowed values:
 | Management and Governance | Endpoint Management                   | Tools that configure and secure access to devices.                                                            |
 | Management and Governance | Observability                         | Monitoring, logging, and tracing of data to track the performance and health of systems.                      |
-| Management and Governance | Support                               | Assistance and expertise supplied by providers.                                                               |
+| Management and Governance | Support                               | Assistance and expertise supplied by service providers.                                                               |
```
```suggestion
| Management and Governance | Support                               | Assistance and expertise supplied by service providers.                                                       |
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/servicesubcategory.md:
```diff
@@ -96,7 +96,7 @@ Allowed values:
 | Migration                 | Other (Migration)                     | Migration services that do not fall into one of the defined subcategories.                                    |
 | Mobile                    | Other (Mobile)                        | All Mobile services.                                                                                          |
-| Multicloud                | Multicloud Integration                | Environments that facilitate consumption of services from multiple cloud providers.                           |
+| Multicloud                | Multicloud Integration                | Environments that facilitate consumption of services from multiple cloud service providers.                           |
```
```suggestion
| Multicloud                | Multicloud Integration                | Environments that facilitate consumption of services from multiple cloud service providers.                   |
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/tags.md:
```diff
@@ -1,35 +1,35 @@
-  * Provider SHOULD publish tag finalization methods and semantics within their respective documentation.
-  * Provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
+  * Service Provider SHOULD publish tag finalization methods and semantics within their respective documentation.
+  * Service Provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
```
```suggestion
  * Service provider SHOULD publish tag finalization methods and semantics within their respective documentation.
  * Service provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/tags.md:
Wondering if all these references should be to data generator instead of service provider...

shawnalpay replied:
@rileyjenk 
I won't block on this, but I do believe this entire section should refer to `data generator` instead of `service provider`, the way we did for allocations -- with the reasoning that `data generator` is the more inclusive group.

shawnalpay commented on specification/datasets/cost_and_usage/columns/tags.md:
```diff
@@ -1,35 +1,35 @@
-  * Provider MUST NOT prefix tag keys when the provider has only one user-defined tag scheme.
-  * Provider MUST NOT allow reserved tag key prefixes to be used as prefixes for any user-defined tag keys within a prefixless user-defined tag scheme.
+  * Service Provider MUST prefix all but one user-defined tag scheme with a predetermined, provider-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the service provider has more than one user-defined tag scheme.
+  * Service Provider MUST NOT prefix tag keys when the service provider has only one user-defined tag scheme.
```
```suggestion
  * Service provider MUST prefix all but one user-defined tag scheme with a predetermined, provider-specified tag key prefix that is unique to each corresponding user-defined tag scheme when the service provider has more than one user-defined tag scheme.
  * Service provider MUST NOT prefix tag keys when the service provider has only one user-defined tag scheme.
```

shawnalpay commented on specification/datasets/cost_and_usage/dataset.md:
```diff
@@ -53,13 +54,14 @@ The specification for the Cost and Usage dataset defines a group of columns that
 | [Pricing Unit](#pricingunit)                                                  | Dimension   | Mandatory     | True         | String    |
-| [Provider](#providername)                                                     | Dimension   | Mandatory     | False        | String    |
-| [Publisher](#publishername)                                                   | Dimension   | Mandatory     | False        | String    |
+| [ProviderName - DEPRECATED](#providername)                                    | Dimension   | Mandatory     | False        | String    |
```
I'm using display name in this table.  So it's just Provider, not Provider Name.
```suggestion
| [Provider - DEPRECATED](#providername)                                        | Dimension   | Mandatory     | False        | String    |
```

shawnalpay commented on specification/datasets/cost_and_usage/dataset.md:
```diff
@@ -44,6 +44,7 @@ The specification for the Cost and Usage dataset defines a group of columns that
 | [Effective Cost](#effectivecost)                                              | Metric      | Mandatory     | False        | Decimal   |
 | [Invoice ID](#invoiceid)                                                      | Dimension   | Recommended   | True         | String    |
 | [Invoice Issuer](#invoiceissuername)                                          | Dimension   | Mandatory     | False        | String    |
+| [Host Provider](#hostprovidername)                                            | Dimension   | Mandatory     | True         | String    |
```
```suggestion
| [Host Provider Name](#hostprovidername)                                       | Dimension   | Mandatory     | True         | String    |
```

shawnalpay commented on specification/metadata/schema/schema.mdpp:
```diff
@@ -6,25 +6,25 @@ The schema metadata object and its content provide information about the structu
 * [FOCUS Version has changed](#focusversionchanged)
 * [Data Generator Version has changed](#schemametadatatofocusdatareference)
-* [Correcting schema metadata errors](#providermetadataerrorcorrection)
+* [Correcting schema metadata errors](#datageneratormetadataerrorcorrection)
```
Leaving myself a note to make sure that this anchor tag resolves.

shawnalpay commented on specification/overview.md:
```diff
@@ -14,7 +14,7 @@ This project is supported by the [FinOps Foundation][FODO]. This work initially
 This specification is designed to be used by three major groups:
 
-* Billing data generators: Infrastructure and services *providers* that bill based on consumption, such as (but not limited to):
+* Billing data generators: Infrastructure and *service providers* that bill based on consumption, such as (but not limited to):
```
```suggestion
* Billing data generators: Entities that present consumption-based billing information related to infrastructure and *service providers*, such as (but not limited to):
```

shawnalpay commented on specification/supported_features/billed_cost_and_invoice_alignment.md:
```diff
@@ -14,7 +14,7 @@ FOCUS data should be consistent with the costs indicated on payable invoices. Th
 ## Supporting Columns
 
-* ProviderName
+* ServiceProviderName
```
Should we just strike this completely?  It's not in the query that follows.

shawnalpay commented on specification/supported_features/location.md:
```diff
@@ -2,7 +2,7 @@
 ## Description
 
-FOCUS provides structured location data through region and availability zone information. By documenting geographic deployment locations, practitioners can organize and analyze costs based on where resources and services are deployed. This standardized location data helps practitioners understand the geographical distribution of infrastructure across providers.
+FOCUS provides structured location data through region and availability zone information. By documenting geographic deployment locations, practitioners can organize and analyze costs based on where resources and services are deployed. This standardized location data helps practitioners understand the geographical distribution of infrastructure across service providers.
```
```suggestion
FOCUS provides structured location data through region and availability zone information. By documenting geographic deployment locations, practitioners can organize and analyze costs based on where resources and services are deployed. This standardized location data helps practitioners understand the geographical distribution of infrastructure across host providers.
```

shawnalpay commented on specification/supported_features/service_provider_services.md:
```diff
@@ -1,8 +1,8 @@
 ## Description
 
-FOCUS supports providers specifying the services and product offerings that they provide their customers that align with the names practitioners are familiar with. This empowers practitioners to analyze cost by service, report service costs by subaccount, forecast based on historical trends by service, and verify accuracy of services charged across providers.
+FOCUS supports service service providers specifying the services and product offerings that they provide their customers that align with the names practitioners are familiar with. This empowers practitioners to analyze cost by service, report service costs by subaccount, forecast based on historical trends by service, and verify accuracy of services charged across service providers.
```
```suggestion
FOCUS supports service providers specifying the services and product offerings that they provide their customers that align with the names practitioners are familiar with. This empowers practitioners to analyze cost by service, report service costs by subaccount, forecast based on historical trends by service, and verify accuracy of services charged across service providers.
```

shawnalpay committed:
alpha sort supported features

shawnalpay reviewed (changes requested):
@rileyjenk 
First of all, KUDOS -- this is a huge amount of work, and I agreed with probably 80% of your recommended changes.

I've left a bunch of changes for you to consider.  We of course need to address these (and any other feedback you receive) before tomorrow.  I know it doesn't leave you a lot of time.  I will make myself available tonight if you have any questions you wish to discuss.

FYI, I directly committed many of the changes related to data generator-calculated split cost allocation per the group's agreement in Oct 29 TF2, to save you some work.

ijurica commented on specification/datasets/cost_and_usage/columns/allocatedresourceid.md:
### Issue
Let's assume both **ServiceProvider** and **DataGenerator** support split cost allocation. If they reuse the same set of `Allocated*` columns, it becomes impossible to distinguish costs split by ServiceProvider from those further split by DataGenerator.

We have explicitly defined separate columns (e.g., `Tags` for the original resource and `AllocatedTags` for allocation) to enable flexibility — the same dataset artifact provides both results:
- **Result with split cost applied**
- **Result with split cost ignored**

As mentioned yesterday, I lean towards **two almost identical but still two distinct sets of columns for each allocation layer**, so that we can ignore one split simply by **excluding its columns from the query**.

If ServiceProvider and DataGenerator were to reuse the same columns, we would lose this capability and only see the combined result of both splits, without a way to isolate one.

### Example – Ignoring DataGenerator split

Assuming distinct columns for ServiceProvider (`SP_AllocatedResourceId`, `SP_AllocatedTags`) and DataGenerator (`DG_AllocatedResourceId`, `DG_AllocatedTags`), we can isolate ServiceProvider split like this:

```sql
SELECT
  ResourceId,
  SP_AllocatedResourceId,
  SUM(EffectiveCost) AS TotalEffectiveCost
FROM focus_data_table
WHERE ChargeCategory = 'Usage'
  AND SP_AllocatedMethodId IS NOT NULL
GROUP BY ResourceId, SP_AllocatedResourceId;
```
### Suggestion

Maintain separate sets of allocation columns. This ensures the dataset supports queries that isolate one allocation layer or combine both when needed.

Long story short - I suggest we:

- Keep `Allocated*` columns and corresponding attributes **tied to **ServiceProvider**.  
- Same goes for `Tags`.

rileyjenk replied:
This is outside of the scope of this change. We have already decided that any data lineage or progeny type feature is outside of scope of 1.3.

shawnalpay replied:
My understanding from the Oct 29 TF2 discussion was that the majority felt it should be `data generator`, because that would be the most inclusive group.  @ijurica, you are right that there is a potentially meaningful difference between the service provider and the data generator when it comes to consuming allocations and tags.  However, we have not agreed as a group that that would be a path forward.

For the time being, I agree that we would take no action on this suggestion for now.  Please review and discuss in Nov 5 TF2 and confirm.  If there is not consensus, then I would recommend using data generator for now, and logging a Feature Request to review this in a future release.

ijurica replied:
Resolving based on the Oct 29 TF2 discussion.

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/availabilityzone.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Apply suggestions from code review

Co-authored-by: Shawn Alpay <shawn@finops.org>

jpradocueva commented:
Approval postponed to no later than November 6.

shawnalpay commented:
I am working on resolving the merge conflicts in this PR to get it to a state we can work on.  Hold please!

shawnalpay committed:
Merge remote-tracking branch 'origin/working_draft' into refactor_entity_name_references

shawnalpay committed:
address merge conflict

shawnalpay committed:
linter fix after repairing merge conflicts

shawnalpay commented on specification/datasets/cost_and_usage/columns/regionname.md:
```diff
@@ -1,12 +1,12 @@
 RegionName adheres to the following requirements:
 
-* RegionName MUST be present in a Cost and Usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports deploying resources or services within a region.
+* RegionName MUST be present in a Cost and Usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the service provider supports deploying resources or services within a region.
```
```suggestion
* RegionName MUST be present in a Cost and Usage [*FOCUS dataset*](#glossary:FOCUS-dataset) when the host provider supports deploying resources or services within a region.
```

shawnalpay commented on specification/datasets/cost_and_usage/columns/tags.md:
```diff
@@ -1,37 +1,37 @@
-  * Provider SHOULD publish tag finalization methods and semantics within their respective documentation.
-  * Provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
+  * Service Provider SHOULD publish tag finalization methods and semantics within their respective documentation.
+  * Service Provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
```
```suggestion
  * Service provider SHOULD publish tag finalization methods and semantics within their respective documentation.
  * Service provider MUST NOT alter tag values unless applying true (boolean) to valueless tags.
```

shawnalpay commented:
Okay @rileyjenk and @ijurica: I've merged everything from working_draft into this PR (there were a lot of conflicts 😅), and I addressed/resolved all the outdated comments I had.  Let's have another look at it now!

ijurica commented on specification/datasets/cost_and_usage/columns/skupricedetails.md:
```diff
@@ -69,7 +69,7 @@ A set of properties of a SKU Price ID which are meaningful and common to all ins
 ### FOCUS-Defined Properties
 
-The following keys should be used when applicable to facilitate cross-SKU and cross-provider queries for the same conceptual property. FOCUS-defined keys will appear in the list below and custom (e.g., provider-defined) keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
+The following keys should be used when applicable to facilitate cross-SKU and cross-service -provider queries for the same conceptual property. FOCUS-defined keys will appear in the list below and custom (e.g., service-provider-defined) keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
```
```suggestion
The following keys should be used when applicable to facilitate cross-SKU and cross-service-provider queries for the same conceptual property. FOCUS-defined keys will appear in the list below and custom (e.g., service-provider-defined) keys will be prefixed with "x_" to make them easy to identify as well as prevent collisions.
```

ijurica commented on specification/datasets/cost_and_usage/columns/pricingunit.md:
```diff
@@ -1,6 +1,6 @@
 # Pricing Unit
 
-The Pricing Unit represents a provider-specified measurement unit for determining unit prices, indicating how the provider rates measured usage and purchase quantities after applying pricing rules like [*block pricing*](#glossary:block-pricing). Common examples include the number of hours for compute appliance runtime (e.g., `Hours`), gigabyte-hours for a storage appliance (e.g., `GB-Hours`), or  ...
+The Pricing Unit represents a service-provider-specified measurement unit for determining unit prices, indicating how the service-provider rates measured usage and purchase quantities after applying pricing rules like [*block pricing*](#glossary:block-pricing). Common examples include the number of hours for compute appliance runtime (e.g., `Hours`), gigabyte-hours for a storage appliance (e.g.,  ...
```
```suggestion
The Pricing Unit represents a service-provider-specified measurement unit for determining unit prices, indicating how the service provider rates measured usage and purchase quantities after applying pricing rules like [*block pricing*](#glossary:block-pricing). Common examples include the number of hours for compute appliance runtime (e.g., `Hours`), gigabyte-hours for a storage appliance (e.g., `GB-Hours`), or an accumulated count of requests for a network appliance or API service (e.g., `1000 Requests`). Pricing Unit complements the [Pricing Quantity](#pricingquantity) metric. Distinct from the [Consumed Unit](#consumedunit), it focuses on pricing and cost, not [*resource*](#glossary:resource) and [*service*](#glossary:service) consumption, often at a coarser granularity.
```

ijurica commented on specification/appendix/metadata_examples/metadata_examples_overview.md:
```diff
@@ -1,8 +1,8 @@
 # Examples: Metadata
 
-The following sections contain examples of metadata provided by a hypothetical FOCUS data provider called ACME to supply the required reference between the [*FOCUS datasets*](#glossary:FOCUS-dataset) and the schema metadata.  Provider implementations will vary on how the metadata is disseminated; however, the provider's chosen metadata delivery approach should be able to support the structure rep ...
+The following sections contain examples of metadata provided by a hypothetical FOCUS data generator called ACME to supply the required reference between the [*FOCUS datasets*](#glossary:FOCUS-dataset) and the schema metadata.  Data Generator implementations will vary on how the metadata is disseminated; however, the data generator's chosen metadata delivery approach should be able to support the  ...
```
```suggestion
The following sections contain examples of metadata provided by a hypothetical FOCUS data generator called ACME to supply the required reference between the [*FOCUS dataset artifacts*](#glossary:dataset-instance-artifact) and the [Metadata](#metadata).  Data Generator implementations will vary on how the metadata is disseminated; however, the data generator's chosen metadata delivery approach should be able to support the structure represented in this example.
```

ijurica commented on specification/appendix/metadata_examples/metadata_examples_overview.md:
```diff
@@ -1,8 +1,8 @@
+The following sections contain examples of metadata provided by a hypothetical FOCUS data generator called ACME to supply the required reference between the [*FOCUS datasets*](#glossary:FOCUS-dataset) and the schema metadata.  Data Generator implementations will vary on how the metadata is disseminated; however, the data generator's chosen metadata delivery approach should be able to support the  ...
 
-In this example, the provider supports delivery of FOCUS data via file export to a data storage system. It uses JSON as the format for providing the metadata. The provider delivers data every 12 hours into a path structure described below:
+In this example, the data generator supports delivery of FOCUS data via file export to a data storage system. It uses JSON as the format for providing the metadata. The data generator delivers data every 12 hours into a path structure described below:
```
```suggestion
In this example, the data generator supports delivery of *FOCUS dataset artifacts* via file export to a data storage system. It uses JSON as the format for providing the Metadata. The data generator delivers data every 12 hours into a path structure described below:
```

rileyjenk replied:
This needs to be refined in a future version.

ijurica commented on specification/appendix/metadata_examples/metadata_examples_overview.md:
```diff
@@ -1,8 +1,8 @@
 
-In this example, the provider supports delivery of FOCUS data via file export to a data storage system. It uses JSON as the format for providing the metadata. The provider delivers data every 12 hours into a path structure described below:
+In this example, the data generator supports delivery of FOCUS data via file export to a data storage system. It uses JSON as the format for providing the metadata. The data generator delivers data every 12 hours into a path structure described below:
 
```
Since we've introduced multiple datasets and the term *dataset artifact*, I suggest we update the tabular path overview.

**NEW:**

| Description                | Path              |
|:---------------------------|:------------------|
| Root location              | `/FOCUS`          |
| Metadata location          | `/FOCUS/metadata` |
| Dataset artifacts location | `/FOCUS/data`     |

**OLD:**

| Type of data        | Path              |
|:--------------------|:------------------|
| Export location     | `/FOCUS`          |
| Metadata location   | `/FOCUS/metadata` |
| Cost data location  | `/FOCUS/data`     |

rileyjenk replied:
The introduction of dataset artifact is both inaccurate and confusing. We need to address is future versions.

ijurica commented on specification/appendix/saas_examples/saas_examples.mdpp:
```diff
@@ -1,6 +1,6 @@
 # Examples: SaaS
 
-The following section contains examples with how SaaS providers may implement the FOCUS specification. SaaS provider implementations will vary on the level of the detail available in their data, contract terms, purchasing options, and other factors.
+The following section contains examples with how SaaS data generators may implement the FOCUS specification. SaaS data generator implementations will vary on the level of the detail available in their data, contract terms, purchasing options, and other factors.
```
```suggestion
The following section contains examples that illustrate how data generators may produce Cost and Usage FOCUS [*dataset artifact*](#glossary:dataset-artifact) related to SaaS products and services. Note that data generator implementations may vary depending on the level of detail available, contract terms, purchasing options, and other factors.
```

rileyjenk replied:
We need to stop going to artifacts as it will be confusing.

ijurica commented on specification/datasets/cost_and_usage/columns/skupriceid.md:
```diff
@@ -1,14 +1,14 @@
 # SKU Price ID
 
-SKU Price ID is a provider-specified unique identifier that represents a specific [*SKU Price*](#glossary:sku-price) associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used or purchased. It serves as a key reference for a *SKU Price* in a [*price list*](#glossary:price-list) published by a provider, allowing practitioners to look up detailed information about the ...
+SKU Price ID is a service-provider-specified unique identifier that represents a specific [*SKU Price*](#glossary:sku-price) associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used or purchased. It serves as a key reference for a *SKU Price* in a [*price list*](#glossary:price-list) published by a service-provider, allowing practitioners to look up detailed infor ...
```
```suggestion
SKU Price ID is a service-provider-specified unique identifier that represents a specific [*SKU Price*](#glossary:sku-price) associated with a [*resource*](#glossary:resource) or [*service*](#glossary:service) used or purchased. It serves as a key reference for a *SKU Price* in a [*price list*](#glossary:price-list) published by a service provider, allowing practitioners to look up detailed information about the *SKU Price*.
```

ijurica commented on specification/metadata/data_generator/data_generator.md:
```diff
@@ -1,6 +1,6 @@
 # Data Generator
 
-Human-readable name of the entity that generated the dataset instance, typically based on data provided by the [*Service Provider*](#serviceprovidername). The Data Generator ensures the technical accuracy and delivery of the data.
+Human-readable name of the entity that generated the dataset instance. The Data Generator ensures the technical accuracy and delivery of the data.
```
I suggest we:

-  **retain the reference to the Service Provider**, as it adds context, clarifies the relationship between the Service Provider and the Data Generator and helps understand their roles, 
- and also explicitly mention the **Metadata** and **dataset instance artifacts**, to reflect the full scope of what the data generator produces

```suggestion
Human-readable name of the entity that generates the [FOCUS Metadata](#metadata) and [*FOCUS dataset instance artifacts*](#glossary:dataset-instance-artifact), typically either the [*service provider*](#serviceprovidername) itself or a separate entity that derives data from information originally supplied by the service provider. The data generator ensures the technical accuracy and delivery of the data.
```

rileyjenk replied:
This add in my opinion is more confusing.

rileyjenk committed:
Apply suggestions from code review

Co-authored-by: Irena Jurica <irena.jurica@neos.hr>
Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/tags.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/tags.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update specification/overview.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update specification/supported_features/location.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/skupricedetails.md

Co-authored-by: Irena Jurica <irena.jurica@neos.hr>

rileyjenk committed:
Update specification/datasets/cost_and_usage/columns/skupriceid.md

Co-authored-by: Irena Jurica <irena.jurica@neos.hr>

rileyjenk committed:
Update specification/supported_features/service_provider_services.md

Co-authored-by: Shawn Alpay <shawn@finops.org>

rileyjenk committed:
Update billed_cost_and_invoice_alignment.md

ijurica commented on specification/data/saas_examples/spend_agreements/saas_spend_agreements_b2.csv:
```diff
@@ -1,15 +1,15 @@
-,0,000-00-000,MyBillingAcct,USD,3/1/26,2/1/26,,,Usage,,Monthly unused fee,One-Time,3/1/26,2/1/26,,,,,,,,,,60,1200,60,Acme Co,60,1200,Standard,0.05,Count,Acme Co,Acme Co,,,,,,Database,AwesomeDB,No-SQL DB,C-004,,,C-004-0,,,
-,0,000-00-000,MyBillingAcct,USD,4/1/26,3/1/26,,,Usage,,End of contract unused fee,One-Time,4/1/26,4/1/25,,,,,,,,,,480,1200,480,Acme Co,480,1200,Standard,0.4,Count,Acme Co,Acme Co,,,,,,Database,AwesomeDB,No-SQL DB,C-004,,,C-004-0,,,
\ No newline at end of file
+AvailabilityZone,BilledCost,BillingAccountId,BillingAccountName,BillingCurrency,BillingPeriodEnd,BillingPeriodStart,CapacityReservationId,CapacityReservationStatus,ChargeCategory,ChargeClass,ChargeDescription,ChargeFrequency,ChargePeriodEnd,ChargePeriodStart,CommitmentDiscountCategory,CommitmentDiscountId,CommitmentDiscountName,CommitmentDiscountQuantity,CommitmentDiscountStatus,CommitmentDiscoun ...
```
```suggestion
AvailabilityZone,BilledCost,BillingAccountId,BillingAccountName,BillingCurrency,BillingPeriodEnd,BillingPeriodStart,CapacityReservationId,CapacityReservationStatus,ChargeCategory,ChargeClass,ChargeDescription,ChargeFrequency,ChargePeriodEnd,ChargePeriodStart,CommitmentDiscountCategory,CommitmentDiscountId,CommitmentDiscountName,CommitmentDiscountQuantity,CommitmentDiscountStatus,CommitmentDiscountType,CommitmentDiscountUnit,ConsumedQuantity,ConsumedUnit,ContractedCost,ContractedUnitPrice,EffectiveCost,InvoiceIssuerName,ListCost,ListUnitPrice,PricingCategory,PricingQuantity,PricingUnit,ServiceProviderName,ProviderName,PublisherName,RegionId,RegionName,ResourceId,ResourceName,ResourceType,ServiceCategory,ServiceName,ServiceSubcategory,SkuId,SkuMeter,SkuPriceDetails,SkuPriceId,SubAccountId,SubAccountName,Tags
```

ijurica commented on specification/data/saas_examples/spend_agreements/saas_spend_agreements_b2.csv:
Note: The current SaaS Examples (15 CSV files in the `saas_examples` folder)) include the following columns:
- InvoiceIssuerName
- ServiceProviderName
- ProviderName
- PublisherName

So, we’ve kept the deprecated columns (`ProviderName`, `PublisherName`) and added `ServiceProviderName`, but not `HostProviderName`.

Should we further refine the examples within this PR, or leave that for the next iteration (v1.4)?

shawnalpay replied:
@ijurica 
We should do this -- but can you please work with @Matt-Cowsert to get a Feature Request created for this?  We are out of time to try and address that as part of 1.3.

ijurica replied:
@shawnalpay @Matt-Cowsert I've created an AI (AI #1584). We can convert it into a Feature Request (FR) if needed.
Resolving this conversation.

ijurica commented on specification/glossary.md:
```diff
@@ -186,9 +186,9 @@ An individual who performs FinOps within an organization to maximize the busines
-<a name="glossary:provider"><b>Provider</b></a>
+<a name="glossary:service provider"><b>Service Provider</b></a>
 
 An entity that made internal or 3rd party resources and/or services available for purchase.
```
```suggestion
An entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase.
```

#### Service Provider Name

- **Definition:** Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase.
- **Description:** The name of the entity that made the *resources* or *services* available for purchase or consumption.
   ⚠️ **Note:** Consider revising the Service Provider Name description!

#### Provider - DEPRECATED

- **Definition:** Provider is the name of the entity that makes the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for purchase.
- **Description:** The name of the entity that made the *resources* or *services* available for purchase.

shawnalpay replied:
@ijurica 
The suggestion as presented here is good with me.  

As to the other points you make: I don't disagree with the proposed changes to the descriptions in the column markdowns, but we need to close on this PR ASAP.

Therefore, please discuss in TF2 today.

* If you can get agreement to do it, then do it live in the meeting and resolve this comment.  
* If you cannot, then resolve this comment and make a note to revisit this editorial change in a future PR.

ijurica commented on specification/supported_features/charge_categorization.md:
We should address Providername in Report on Corrections

### Report on Corrections

```sql
SELECT
  ProviderName,
  BillingAccountId,
  ChargeCategory,
  ServiceCategory,
  ServiceName,
  SUM(BilledCost) AS TotalBilledCost
FROM focus_data_table
WHERE BillingPeriodStart >= ? AND BillingPeriodEnd < ?
  AND ChargeClass = 'Correction'
GROUP BY
  ProviderName,
  BillingAccountId,
  ChargeCategory,
  ServiceCategory,
  ServiceName
```

shawnalpay replied:
@ijurica 
I'm not following what the change is here. Please leave a suggestion in the appropriate place so @rileyjenk can more easily review and commit.

ijurica replied:
I couldn’t provide suggestions on those lines (that SQL Query), so I’ve committed the fix directly to the branch.

shawnalpay replied:
Looks good to me!  I'm good to resolve this one.

jpradocueva committed:
replace Provider with Service Provider

jpradocueva committed:
resolve issues with naming service provider

jpradocueva reviewed (approved):
The diagram has been updated to reflect "Service Provider"

ijurica commented on specification/supported_features/marketplace_purchases.md:
```diff
@@ -27,7 +27,7 @@ FROM focus_data_table
 WHERE ProviderName = '<Example SaaS Provider>'
```
```suggestion
WHERE ServiceProviderName = '<Example SaaS Provider>'
```

ijurica replied:
I've committed this directly. Resolving.

ijurica commented on specification/supported_features/cost_comparison.md:
Line 58: Replace `ProviderName` with `ServiceProviderName`

shawnalpay replied:
@ijurica 
Please leave a suggestion in the appropriate places so @rileyjenk can more easily review and commit.

ijurica replied:
I couldn’t provide suggestions on that line, so I’ve committed the fix directly to the branch.

Note: Syntax issue (missing comma) in this SQL Query addressed in a separate PR

shawnalpay replied:
Looks good to me!  I'm good with resolving this one.

ijurica commented on specification/supported_features/commit_usage_and_under_usage.md:
Replace `ProviderName` with `ServiceProviderName` in:
- line 49
- line 59

shawnalpay replied:
@ijurica 
Please leave a suggestion in the appropriate places so @rileyjenk can more easily review and commit.

ijurica replied:
I couldn’t provide suggestions on those lines, so I’ve committed the fix directly to the branch.

shawnalpay replied:
Looks good to me!  I'm good with resolving this one.

shawnalpay commented:
@ijurica Please leave specific suggestions so @rileyjenk can more easily incorporate your feedback.

shawnalpay reviewed (changes requested):
@rileyjenk @ijurica (cc @Matt-Cowsert)
I have commented or 👍'd in every outstanding thread.  Please plan to discuss and resolve these items in Nov 5 TF2.

ijurica commented:
> @ijurica Please leave specific suggestions so @rileyjenk can more easily incorporate your feedback.

@shawnalpay @rileyjenk I wasn’t able to comment directly on the mentioned lines in the PR, so I left suggestions referencing the specific lines where the term should be replaced. 
Alternatively, I can make the changes and commit them directly if that works better.

shawnalpay commented:
> > @ijurica Please leave specific suggestions so @rileyjenk can more easily incorporate your feedback.
> 
> @shawnalpay @rileyjenk I wasn’t able to comment directly on the mentioned lines in the PR, so I left suggestions referencing the specific lines where the term should be replaced. Alternatively, I can make the changes and commit them directly if that works better.

Ahh, I see.  Sorry, I should have realized that.

I just looked at them again.  Those are truly cleanup activities.  I say just go ahead and do a commit.  Thank you!

ijurica committed:
fixing participating entities in Supported Feature SQL Queries

ijurica committed:
fixing participating entities in Supported Feature SQL Queries

ijurica committed:
fixing participating entities in Supported Feature SQL Queries

ijurica commented:
In the specification we currently have:

- **Data Generator-Calculated Split Cost Allocation Handling** - attribute

- **Data Generator-Calculated Split Cost Allocation** - supported feature

During the renaming of these elements, a few inconsistencies appeared in the specification (wrong titles and/or anchors).

@kk09v @qquigley @shawnalpay 
In one of the commits in this PR, all links were updated to point to the `Data Generator-Calculated Split Cost Allocation Handling` attribute, **but we should confirm if that’s correct**. Optionally, some of those links could instead point to the Supported Feature, if intended.

Note that by convention, we typically use:

```` markdown
[Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling)
````

meaning the link text shows the supported feature name, while the anchor (the part after #) points to the attribute section.

---

#### Tabular Overview

| File | Reference |
|------|------------|
| \glossary.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \glossary.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \glossary.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\dataset.md | [DataGeneratorCalculatedSplitCostAllocationHandling](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\columns\allocatedmethoddetails.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\columns\allocatedmethodid.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\columns\allocatedresourceid.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\columns\allocatedresourcename.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \datasets\cost_and_usage\columns\allocatedtags.md | [Data Generator-Calculated Split Cost Allocation](#datagenerator-calculatedsplitcostallocationhandling) |
| \guidelines\normative-requirements-guidelines.md | [DataGeneratorCalculatedSplitCostAllocationHandling](#datagenerator-calculatedsplitcostallocationhandling) |

ijurica committed:
fixing references to Data Generator-Calculated Split Cost Allocation Handling

Matt-Cowsert committed:
Update specification/data/saas_examples/spend_agreements/saas_spend_agreements_b2.csv

Co-authored-by: Irena Jurica <irena.jurica@neos.hr>

rileyjenk committed:
Update specification/glossary.md

Co-authored-by: Irena Jurica <irena.jurica@neos.hr>

rileyjenk committed:
Removing consume and using the same wording from the description.

Matt-Cowsert committed:
Fix formatting issues in glossary entries

linking glossary from the attribute to the supported feature for split cost allocation

rileyjenk committed:
Updating Tags column to reference data generators

jpradocueva commented:
Approved by the group without objections or abstentions during the November 6 Members’ call.
